aten/src/ATen/native/MaxPooling.cpp

```diff
@@ -107,8 +98,14 @@ Tensor max_pool1d(
     return std::get<0>(at::max_pool1d_with_indices(
         self, kernel_size, stride, padding, dilation, ceil_mode));
   }
-  return max_pool1d_impl(
-      self, kernel_size, stride, padding, dilation, ceil_mode);
+
+  Tensor result = [&]() {
```

Comment:
TODO(rzou): check history

Reply:
Motivation for special kernel on CPU.
PR https://github.com/pytorch/pytorch/pull/43745 description has the relevant benchmark.

Reply:
Thanks for finding it. It seems fine to me that we are turning this into an operator